In [1]:
import sqlite3 as sql
import pandas as pd,numpy as np

In [2]:
db=sql.connect("imdb.db")

In [71]:
#db1="csv1.db"
db

# Phase 2

#### Make two tables and corresponding columns provided in the above CSV files in SQLite DBMS. Insert all data of each CSV file in each of the created tables. Now start querying the table(s) in the  SQL workbench / SQLite database :

##### Table 1: Sno, MovieName, Director Name, Duration, genre, ratings

1)  Display all the details of movies created by directors Christopher and Matt Reeves.

2) Display all the details of movies with a duration of 140 minutes to 190 minutes.

3) Display all details of movies with ratings above 7 in ascending order.

4) Display all movie names in descending order.

5) Display movie name starts with ‘P’ and their rating is greater than 7.

## Display all the details of movies created by directors Christopher and Matt Reeves.

In [3]:
# displaying all the records 
query="select * from Table2"

In [4]:
df=pd.read_sql_query(query,db)

In [5]:
df

movie_name              star1  \
0                                       The Dark Knight     Christian Bale   
1         The Lord of the Rings: The Return of the King        Elijah Wood   
2                   Spider-Man: Across the Spider-Verse        Kemp Powers   
3                                             Inception  Leonardo DiCaprio   
4     The Lord of the Rings: The Fellowship of the Ring        Elijah Wood   
...                                                 ...                ...   
1695                                    Howard the Duck      Natsuki Hanae   
1696                          Kisi Ka Bhai Kisi Ki Jaan     Manoj Bajpayee   
1697                                         The Spirit            Prabhas   
1698                                        Zoolander 2     Toshirô Mifune   
1699                                           Snatched      Vicky Kaushal   

                     star2      votes     genre1     genre2     genre3  \
0             Heath Ledger  2,741,650     Action      Crime      Drama   
1          Viggo Mortensen  1,899,228     Action  Adventure      Drama   
2       Justin K. Thompson    186,212  Animation     Action  Adventure   
3     Joseph Gordon-Levitt  2,432,910     Action  Adventure     Sci-Fi   
4             Ian McKellen  1,927,643     Action  Adventure      Drama   
...                    ...        ...        ...        ...        ...   
1695            Akari Kitô     38,865     Action     Comedy      Drama   
1696   Nawazuddin Siddiqui     62,188     Action      Crime    Fantasy   
1697        Rana Daggubati     73,346     Action  Adventure     Comedy   
1698           Eijirô Tôno     36,636     Action  Adventure     Comedy   
1699          Paresh Rawal     32,861     Action  Adventure     Horror   

      popularity certification     gross  
0            9.0            UA  $2741690  
1            9.0             U  $1899240  
2            8.9             U   $186260  
3            8.8            UA  $2432950  
4            8.8             U  $1927658  
...          ...           ...       ...  
1695         4.7            UA    $66121  
1696         4.7         PG-13    $99908  
1697         4.7             U   $108098  
1698         4.7             R   $126747  
1699         4.7             U    $68321  

[1700 rows x 10 columns]

# query1

In [20]:
#directors = ['Christopher Nolan', 'Matt Reeves']
select_query = "SELECT * FROM Table1 WHERE director IN ('Christopher Nolan', 'Matt Reeves')"
df=pd.read_sql_query(select_query,db)

In [21]:
df

S.no                         Movie_name           director duration  \
0     1                    The Dark Knight  Christopher Nolan  152 min   
1     4                          Inception  Christopher Nolan  148 min   
2    22              The Dark Knight Rises  Christopher Nolan  164 min   
3    40                      Batman Begins  Christopher Nolan  140 min   
4   131  Everything Everywhere All at Once        Matt Reeves  139 min   
5   132                         The Batman  Christopher Nolan  176 min   
6   230               The Raid: Redemption        Matt Reeves  101 min   
7   309                Law Abiding Citizen        Matt Reeves  109 min   
8   322                       Bullet Train  Christopher Nolan  127 min   
9   515        Sicario: Day of the Soldado        Matt Reeves  122 min   

   movie_year  rating  
0        2008     9.0  
1        2010     8.8  
2        2012     8.4  
3        2005     8.2  
4        2022     7.8  
5        2022     7.8  
6        2011     7.6  
7        2009     7.4  
8        2022     7.3  
9        2018     7.0

### the movies directed by christoper noaln is 6 and matt reeves 4 total is 10 movies 

# query2

### 2) Display all the details of movies with a duration of 140 minutes to 190 minutes.

In [22]:
select_query = "SELECT * FROM Table1 WHERE duration BETWEEN 140  and 190 "
df=pd.read_sql_query(select_query,db)

In [23]:
df

S.no                                         Movie_name  \
0        1                                    The Dark Knight   
1        3                Spider-Man: Across the Spider-Verse   
2        4                                          Inception   
3        5  The Lord of the Rings: The Fellowship of the Ring   
4        6              The Lord of the Rings: The Two Towers   
..     ...                                                ...   
195  1,665                                  Bachchhan Paandey   
196  1,675                                            Dilwale   
197  1,684                                     Happy New Year   
198  1,693                                          Shamshera   
199  1,696                          Kisi Ka Bhai Kisi Ki Jaan   

               director duration movie_year  rating  
0     Christopher Nolan  152 min       2008     9.0  
1    Joaquim Dos Santos  140 min       2023     8.9  
2     Christopher Nolan  148 min       2010     8.8  
3         Peter Jackson  178 min       2001     8.8  
4         Peter Jackson  179 min       2002     8.8  
..                  ...      ...        ...     ...  
195     Elizabeth Banks  147 min       2022     5.1  
196    Barry Sonnenfeld  158 min       2015     5.0  
197        Frank Coraci  180 min   I) (2014     4.9  
198          Rob Bowman  178 min       2022     4.8  
199        Frank Miller  145 min       2023     4.7  

[200 rows x 6 columns]

### here is 200 movies between 140 to 190 duration

# query3

### 3) Display all details of movies with ratings above 7 in ascending order.

In [66]:
select_query="select * from Table1 where rating >7 order by rating asc"
df=pd.read_sql_query(select_query,db)

In [67]:
df

index S.no                                         Movie_name  \
0      433  433                                Mission: Impossible   
1      434  434                                      Extraction II   
2      435  435                                          The Mummy   
3      436  436                                       Man of Steel   
4      437  437                                               Prey   
..     ...  ...                                                ...   
489      4    5  The Lord of the Rings: The Fellowship of the Ring   
490      5    6              The Lord of the Rings: The Two Towers   
491      2    3                Spider-Man: Across the Spider-Verse   
492      0    1                                    The Dark Knight   
493      1    2      The Lord of the Rings: The Return of the King   

               director duration movie_year  rating  
0          Sam Hargrave  110 min       1996     7.1  
1       Stephen Sommers  122 min       2023     7.1  
2           Zack Snyder  124 min       1999     7.1  
3      Dan Trachtenberg  143 min       2013     7.1  
4           Ben Stiller  100 min   I) (2022     7.1  
..                  ...      ...        ...     ...  
489       Peter Jackson  178 min       2001     8.8  
490       Peter Jackson  179 min       2002     8.8  
491  Joaquim Dos Santos  140 min       2023     8.9  
492   Christopher Nolan  152 min       2008     9.0  
493       Peter Jackson  201 min       2003     9.0  

[494 rows x 7 columns]

### here is 494 rows with rating higher than or above 7 in ascending order

# query4 

### 4) Display all movie names in descending order.

In [72]:
select_query="select Movie_name from Table1 order by Movie_name desc "
df=pd.read_sql_query(select_query,db)

In [73]:
df

Movie_name
0                       Æon Flux
1     xXx: Return of Xander Cage
2                            xXx
3                    Zoolander 2
4         Zombieland: Double Tap
...                          ...
1695                   12 Strong
1696                   12 Rounds
1697                   10,000 BC
1698                         '71
1699                      #Alive

[1700 rows x 1 columns]

# query 4 

### 5) Display movie name starts with ‘P’ and their rating is greater than 7

In [92]:
#conn = sqlite3.connect('csv1.db')
#cursor = conn.cursor()
#prefix = 'P'
#min_rating = 7.0
#
#select_query = "SELECT Movie_name FROM Table1 WHERE Movie_name LIKE ? AND rating > ?"
#cursor.execute(select_query, (prefix + '%', min_rating))
#
#movie_names = cursor.fetchall()
#
#for name in movie_names:
#    print(name[0])


In [98]:
select_query="select Movie_name,rating from Table1 where Movie_name like 'P%' and rating >7"
df=pd.read_sql_query(select_query,db)

In [99]:
df

Movie_name  rating
0                                   Princess Mononoke     8.3
1                                    Paan Singh Tomar     8.2
2   Pirates of the Caribbean: The Curse of the Bla...     8.1
3                                            Predator     7.8
4                             Ponniyin Selvan: Part I     7.6
5                           Pushpa: The Rise - Part 1     7.6
6                                        Police Story     7.5
7                                      Predestination     7.4
8          Pirates of the Caribbean: Dead Man's Chest     7.3
9                                        Patriots Day     7.3
10                                        Point Break     7.2
11                                               Prey     7.1
12           Pirates of the Caribbean: At World's End     7.1
13                                            Payback     7.1

##### here is 13 movies starts with name P with rating above 7

# Table 2 

##### Table 2: Movie Name, stars, votes, Genre, Gross collection, popularity, Certification

1) Display all movie names with star Arnold Schwarzenegger in ascending order.

2) Display all details of the movie with the highest number of votes.

3) Display movie names with gross collections in descending order.

4) Display the gross collection of movies with the star Arnold.

5) Display all details of movies with comedy and action genres.

##  before runnign the query we have to connect our table2 with sqlite3

In [24]:
select_query="select * from Table2 "
df=pd.read_sql_query(select_query,db)

In [25]:
df

movie_name              star1  \
0                                       The Dark Knight     Christian Bale   
1         The Lord of the Rings: The Return of the King        Elijah Wood   
2                   Spider-Man: Across the Spider-Verse        Kemp Powers   
3                                             Inception  Leonardo DiCaprio   
4     The Lord of the Rings: The Fellowship of the Ring        Elijah Wood   
...                                                 ...                ...   
1695                                    Howard the Duck      Natsuki Hanae   
1696                          Kisi Ka Bhai Kisi Ki Jaan     Manoj Bajpayee   
1697                                         The Spirit            Prabhas   
1698                                        Zoolander 2     Toshirô Mifune   
1699                                           Snatched      Vicky Kaushal   

                     star2      votes     genre1     genre2     genre3  \
0             Heath Ledger  2,741,650     Action      Crime      Drama   
1          Viggo Mortensen  1,899,228     Action  Adventure      Drama   
2       Justin K. Thompson    186,212  Animation     Action  Adventure   
3     Joseph Gordon-Levitt  2,432,910     Action  Adventure     Sci-Fi   
4             Ian McKellen  1,927,643     Action  Adventure      Drama   
...                    ...        ...        ...        ...        ...   
1695            Akari Kitô     38,865     Action     Comedy      Drama   
1696   Nawazuddin Siddiqui     62,188     Action      Crime    Fantasy   
1697        Rana Daggubati     73,346     Action  Adventure     Comedy   
1698           Eijirô Tôno     36,636     Action  Adventure     Comedy   
1699          Paresh Rawal     32,861     Action  Adventure     Horror   

      popularity certification     gross  
0            9.0            UA  $2741690  
1            9.0             U  $1899240  
2            8.9             U   $186260  
3            8.8            UA  $2432950  
4            8.8             U  $1927658  
...          ...           ...       ...  
1695         4.7            UA    $66121  
1696         4.7         PG-13    $99908  
1697         4.7             U   $108098  
1698         4.7             R   $126747  
1699         4.7             U    $68321  

[1700 rows x 10 columns]

# query1

#### 1) Display all movie names with star Arnold Schwarzenegger in ascending order.

In [120]:
select_query="select movie_name,star1 from Table2 where star1 in ('Arnold Schwarzenegger') "
df=pd.read_sql_query(select_query,db)

In [121]:
df

movie_name                  star1
0       Terminator 2: Judgment Day  Arnold Schwarzenegger
1   Guardians of the Galaxy Vol. 3  Arnold Schwarzenegger
2                        Star Trek  Arnold Schwarzenegger
3        Aguirre, the Wrath of God  Arnold Schwarzenegger
4                      Baby Driver  Arnold Schwarzenegger
5                       Collateral  Arnold Schwarzenegger
6              Law Abiding Citizen  Arnold Schwarzenegger
7                   The Negotiator  Arnold Schwarzenegger
8              In the Line of Fire  Arnold Schwarzenegger
9                       Real Steel  Arnold Schwarzenegger
10                         Super 8  Arnold Schwarzenegger
11                          Sultan  Arnold Schwarzenegger
12                        Repo Man  Arnold Schwarzenegger
13                    Ghostbusters  Arnold Schwarzenegger
14                            Tron  Arnold Schwarzenegger
15                     Red Sparrow  Arnold Schwarzenegger
16                 Ghostbusters II  Arnold Schwarzenegger
17                       12 Strong  Arnold Schwarzenegger
18               Batman: The Movie  Arnold Schwarzenegger
19                      Death Race  Arnold Schwarzenegger
20                      Red Notice  Arnold Schwarzenegger
21                   Osmosis Jones  Arnold Schwarzenegger
22                 The Devil's Own  Arnold Schwarzenegger
23                      Blown Away  Arnold Schwarzenegger
24                I Am Number Four  Arnold Schwarzenegger
25                         Thunivu  Arnold Schwarzenegger
26                   The Condemned  Arnold Schwarzenegger
27                        Daylight  Arnold Schwarzenegger
28                    Hard to Kill  Arnold Schwarzenegger
29                     The Do-Over  Arnold Schwarzenegger
30                  Chain Reaction  Arnold Schwarzenegger
31                    Babylon A.D.  Arnold Schwarzenegger
32         The Karate Kid Part III  Arnold Schwarzenegger
33                Agent Cody Banks  Arnold Schwarzenegger

##### here is 33 movies in which Arnold Schwarzenegger is played chracter of star

# query 2

#### 2) Display all details of the movie with the highest number of votes.

In [26]:
select_query="SELECT * FROM Table2 ORDER BY Votes DESC LIMIT 1"
df=pd.read_sql_query(select_query,db)

In [27]:
df

movie_name        star1               star2   votes  genre1  genre2  \
0  Firestarter  Kemp Powers  Justin K. Thompson  99,918  Action  Sci-Fi   

     genre3  popularity certification    gross  
0  Thriller         6.0         PG-13  $186260

# query3

#### 3) Display movie names with gross collections in descending order.

In [127]:
select_query="select movie_name,gross from Table2 order by gross  desc"
df=pd.read_sql_query(select_query,db)

In [129]:
df.head(14)

movie_name   gross
0                                  Gangs of Wasseypur  $99908
1                                         Wolfwalkers  $99908
2                          Captain America: Civil War  $99908
3                                            Scarface  $99908
4                                            Cell 211  $99908
5                   John Wick: Chapter 3 - Parabellum  $99908
6                                            Megamind  $99908
7   Batman v Superman: Dawn of Justice (Ultimate E...  $99908
8                                      Batman Returns  $99908
9                                    Independence Day  $99908
10                               Ip Man 4: The Finale  $99908
11                          Atlantis: The Lost Empire  $99908
12                                            Hellboy  $99908
13                                      Violent Night  $99908

## query4

#### 4) Display the gross collection of movies with the star Arnold.

In [144]:
select_query="select gross,movie_name from Table2 where star1 in ('Arnold Schwarzenegger') "
df=pd.read_sql_query(select_query,db)

In [145]:
df

gross                      movie_name
0   $1131213      Terminator 2: Judgment Day
1   $1131213  Guardians of the Galaxy Vol. 3
2   $1131213                       Star Trek
3   $1131213       Aguirre, the Wrath of God
4   $1131213                     Baby Driver
5   $1131213                      Collateral
6   $1131213             Law Abiding Citizen
7   $1131213                  The Negotiator
8   $1131213             In the Line of Fire
9   $1131213                      Real Steel
10  $1131213                         Super 8
11  $1131213                          Sultan
12  $1131213                        Repo Man
13  $1131213                    Ghostbusters
14  $1131213                            Tron
15  $1131213                     Red Sparrow
16  $1131213                 Ghostbusters II
17  $1131213                       12 Strong
18  $1131213               Batman: The Movie
19  $1131213                      Death Race
20  $1131213                      Red Notice
21  $1131213                   Osmosis Jones
22  $1131213                 The Devil's Own
23  $1131213                      Blown Away
24  $1131213                I Am Number Four
25  $1131213                         Thunivu
26  $1131213                   The Condemned
27  $1131213                        Daylight
28  $1131213                    Hard to Kill
29  $1131213                     The Do-Over
30  $1131213                  Chain Reaction
31  $1131213                    Babylon A.D.
32  $1131213         The Karate Kid Part III
33  $1131213                Agent Cody Banks

## query5

#### 5) Display all details of movies with comedy and action genres.

In [41]:
select_query="select movie_name,genre1,genre2 from Table2 where 'Comedy' in (genre1, genre2, genre3) AND 'Action' IN (genre1, genre2, genre3)"
df=pd.read_sql_query(select_query,db)

In [42]:
df

movie_name  genre1     genre2
0                                Gangs of Wasseypur  Action     Comedy
1                                  K.G.F: Chapter 1  Action     Comedy
2                                  Sound of Freedom  Action  Adventure
3                          How to Train Your Dragon  Action     Comedy
4    Neon Genesis Evangelion: The End of Evangelion  Action  Adventure
..                                              ...     ...        ...
345                                     Double Team  Action  Adventure
346                                         Elektra  Action  Adventure
347                                 Howard the Duck  Action     Comedy
348                                      The Spirit  Action  Adventure
349                                     Zoolander 2  Action  Adventure

[350 rows x 3 columns]

#### here is 350 movies with genre action and comedy

# subqueries from both of the tables 

#### Make subquery :

1) Display all details from both tables where movie names are the same.

2) Display all movie names, Director, ratings, and gross collection where the genre is action.

3) Display all details from both tables with the highest gross collection.

4) Display all details from both tables with the highest ratings

5) Display all details from both tables with the lowest gross collection and lowest ratings

### 1) Display all details from both tables where movie names are the same.

In [3]:
select_query ='''select Table2.movie_name,Table2.star1,Table2.star2,Table2.votes,Table2.genre1,Table2.genre2,Table2.genre3,Table2.popularity,Table2.certification,Table2.gross ,Table1.director,Table1.duration,Table1.movie_year,Table1.rating from Table1
inner join Table2 
on Table1.movie_name=Table2.movie_name; '''
df = pd.read_sql_query(select_query, db)

In [4]:
df

movie_name              star1  \
0                                       The Dark Knight     Christian Bale   
1         The Lord of the Rings: The Return of the King        Elijah Wood   
2                   Spider-Man: Across the Spider-Verse        Kemp Powers   
3                                             Inception  Leonardo DiCaprio   
4     The Lord of the Rings: The Fellowship of the Ring        Elijah Wood   
...                                                 ...                ...   
1757                                    Howard the Duck      Natsuki Hanae   
1758                          Kisi Ka Bhai Kisi Ki Jaan     Manoj Bajpayee   
1759                                         The Spirit            Prabhas   
1760                                        Zoolander 2     Toshirô Mifune   
1761                                           Snatched      Vicky Kaushal   

                     star2      votes     genre1     genre2     genre3  \
0             Heath Ledger  2,741,650     Action      Crime      Drama   
1          Viggo Mortensen  1,899,228     Action  Adventure      Drama   
2       Justin K. Thompson    186,212  Animation     Action  Adventure   
3     Joseph Gordon-Levitt  2,432,910     Action  Adventure     Sci-Fi   
4             Ian McKellen  1,927,643     Action  Adventure      Drama   
...                    ...        ...        ...        ...        ...   
1757            Akari Kitô     38,865     Action     Comedy      Drama   
1758   Nawazuddin Siddiqui     62,188     Action      Crime    Fantasy   
1759        Rana Daggubati     73,346     Action  Adventure     Comedy   
1760           Eijirô Tôno     36,636     Action  Adventure     Comedy   
1761          Paresh Rawal     32,861     Action  Adventure     Horror   

      popularity certification     gross            director duration  \
0            9.0            UA  $2741690   Christopher Nolan  152 min   
1            9.0             U  $1899240       Peter Jackson  201 min   
2            8.9             U   $186260  Joaquim Dos Santos  140 min   
3            8.8            UA  $2432950   Christopher Nolan  148 min   
4            8.8             U  $1927658       Peter Jackson  178 min   
...          ...           ...       ...                 ...      ...   
1757         4.7            UA    $66121        Farhad Samji  110 min   
1758         4.7         PG-13    $99908        Frank Miller  145 min   
1759         4.7             U   $108098         Ben Stiller  103 min   
1760         4.7             R   $126747     Jonathan Levine  101 min   
1761         4.7             U    $68321    Dwight H. Little   90 min   

     movie_year  rating  
0          2008     9.0  
1          2003     9.0  
2          2023     8.9  
3          2010     8.8  
4          2001     8.8  
...         ...     ...  
1757       1986     4.7  
1758       2023     4.7  
1759       2008     4.7  
1760       2016     4.7  
1761       2017     4.7  

[1762 rows x 14 columns]

### 2) Display all movie names, Director, ratings, and gross collection where the genre is action.

In [50]:
select_query ='''SELECT Table1.movie_name, Table1.director,Table2.genre1, Table1.rating, Table2.gross
FROM Table1
JOIN Table2 ON Table1.movie_name = Table2.movie_name
WHERE Table2.genre1 = 'Action' '''

df = pd.read_sql_query(select_query, db)

In [51]:
df

movie_name           director  \
0                                       The Dark Knight  Christopher Nolan   
1         The Lord of the Rings: The Return of the King      Peter Jackson   
2                                             Inception  Christopher Nolan   
3     The Lord of the Rings: The Fellowship of the Ring      Peter Jackson   
4                 The Lord of the Rings: The Two Towers      Peter Jackson   
...                                                 ...                ...   
1684                                    Howard the Duck       Farhad Samji   
1685                          Kisi Ka Bhai Kisi Ki Jaan       Frank Miller   
1686                                         The Spirit        Ben Stiller   
1687                                        Zoolander 2    Jonathan Levine   
1688                                           Snatched   Dwight H. Little   

      genre1  rating     gross  
0     Action     9.0  $2741690  
1     Action     9.0  $1899240  
2     Action     8.8  $2432950  
3     Action     8.8  $1927658  
4     Action     8.8  $1713981  
...      ...     ...       ...  
1684  Action     4.7    $66121  
1685  Action     4.7    $99908  
1686  Action     4.7   $108098  
1687  Action     4.7   $126747  
1688  Action     4.7    $68321  

[1689 rows x 5 columns]

## 3)Display all details from both tables with the highest gross collection. 

In [7]:
select_query='''select Table1.movie_name,Table2.star1,Table2.star2,Table2.votes,Table2.genre1,Table2.popularity,Table2.certification,Table2.gross ,Table1.director,Table1.duration,Table1.movie_year,Table1.rating from Table1
inner join Table2 
on Table1.movie_name=Table2.movie_name
where gross= (select max(gross) from Table2)'''

df=pd.read_sql_query(select_query,db)

In [8]:
df

movie_name           star1  \
0                                  Gangs of Wasseypur  Manoj Bajpayee   
1                                         Wolfwalkers  Manoj Bajpayee   
2                          Captain America: Civil War  Manoj Bajpayee   
3                                            Scarface  Manoj Bajpayee   
4                                            Cell 211  Manoj Bajpayee   
5                   John Wick: Chapter 3 - Parabellum  Manoj Bajpayee   
6                                            Megamind  Manoj Bajpayee   
7   Batman v Superman: Dawn of Justice (Ultimate E...  Manoj Bajpayee   
8                                      Batman Returns  Manoj Bajpayee   
9                                    Independence Day  Manoj Bajpayee   
10                               Ip Man 4: The Finale  Manoj Bajpayee   
11                          Atlantis: The Lost Empire  Manoj Bajpayee   
12                                            Hellboy  Manoj Bajpayee   
13                                            Hellboy  Manoj Bajpayee   
14                                      Violent Night  Manoj Bajpayee   
15                               Machine Gun Preacher  Manoj Bajpayee   
16                                   The 13th Warrior  Manoj Bajpayee   
17                                     Fast & Furious  Manoj Bajpayee   
18                          The Spiderwick Chronicles  Manoj Bajpayee   
19                                            Savages  Manoj Bajpayee   
20                                        Night Watch  Manoj Bajpayee   
21                                         Entrapment  Manoj Bajpayee   
22                          Jack Ryan: Shadow Recruit  Manoj Bajpayee   
23                                        Tower Heist  Manoj Bajpayee   
24                                            RoboCop  Manoj Bajpayee   
25                                            RoboCop  Manoj Bajpayee   
26                                     Mercury Rising  Manoj Bajpayee   
27                                      Transporter 3  Manoj Bajpayee   
28                                           Iron Sky  Manoj Bajpayee   
29                    Pride and Prejudice and Zombies  Manoj Bajpayee   
30                           Buffy the Vampire Slayer  Manoj Bajpayee   
31          Fantastic Four: Rise of the Silver Surfer  Manoj Bajpayee   
32                            The Cloverfield Paradox  Manoj Bajpayee   
33                                            Dhoom 3  Manoj Bajpayee   
34                                         Alex Cross  Manoj Bajpayee   
35                          Kisi Ka Bhai Kisi Ki Jaan  Manoj Bajpayee   

                  star2    votes     genre1  popularity certification   gross  \
0   Nawazuddin Siddiqui   99,908     Action         8.2             A  $99908   
1   Nawazuddin Siddiqui  128,147     Action         8.0            UA  $99908   
2   Nawazuddin Siddiqui  617,299     Action         7.8             A  $99908   
3   Nawazuddin Siddiqui   42,746     Action         7.7          None  $99908   
4   Nawazuddin Siddiqui   25,597  Animation         7.6             U  $99908   
5   Nawazuddin Siddiqui  687,358     Action         7.4             U  $99908   
6   Nawazuddin Siddiqui  307,406     Action         7.3            18  $99908   
7   Nawazuddin Siddiqui  173,711     Action         7.2             A  $99908   
8   Nawazuddin Siddiqui  345,203     Action         7.1             A  $99908   
9   Nawazuddin Siddiqui  129,353     Action         7.0             A  $99908   
10  Nawazuddin Siddiqui  106,127     Action         7.0             A  $99908   
11  Nawazuddin Siddiqui   46,311     Action         6.9             A  $99908   
12  Nawazuddin Siddiqui  129,229     Action         6.8             A  $99908   
13  Nawazuddin Siddiqui  129,229     Action         6.8             A  $99908   
14  Nawazuddin Siddiqui  480,125     Action         6.7            UA  $99908   
15  Nawazuddin Siddiqui   26,942     Action  

### Display all details from both tables with the highest ratings

In [9]:
select_query='''select Table1.movie_name,Table2.star1,Table2.star2,Table2.votes,Table2.genre1,Table2.popularity,Table2.certification,Table2.gross ,Table1.director,Table1.duration,Table1.movie_year,Table1.rating from Table1
inner join Table2 
on Table1.movie_name=Table2.movie_name
where rating= (select max(rating) from Table1)'''
df=pd.read_sql_query(select_query,db)

In [10]:
df

movie_name           star1  \
0                                The Dark Knight  Christian Bale   
1  The Lord of the Rings: The Return of the King     Elijah Wood   

             star2      votes  genre1  popularity certification     gross  \
0     Heath Ledger  2,741,650  Action         9.0            UA  $2741690   
1  Viggo Mortensen  1,899,228  Action         9.0             U  $1899240   

            director duration  movie_year  rating  
0  Christopher Nolan  152 min        2008     9.0  
1      Peter Jackson  201 min        2003     9.0

### Display all details from both tables with the lowest gross collection and lowest ratings

In [11]:
select_query='''select Table1.movie_name,Table2.star1,Table2.star2,Table2.votes,Table2.genre1,Table2.popularity,Table2.certification,Table2.gross ,Table1.director,Table1.duration,Table1.movie_year,Table1.rating from Table1
inner join Table2 
on Table1.movie_name=Table2.movie_name
where rating= (select min(rating) from Table1) or gross=(select min(gross) from Table2)'''
df=pd.read_sql_query(select_query,db)

In [12]:
df

movie_name            star1  \
0                  Raiders of the Lost Ark    Harrison Ford   
1                                     Rush    Harrison Ford   
2                          Children of Men    Harrison Ford   
3                            Minnal Murali    Harrison Ford   
4   Harry Potter and the Half-Blood Prince    Harrison Ford   
5                              Source Code    Harrison Ford   
6                           Black Dynamite    Harrison Ford   
7                                   Nikita    Harrison Ford   
8                          The Italian Job    Harrison Ford   
9                          Kung Fu Panda 3    Harrison Ford   
10                         The Italian Job    Harrison Ford   
11                             Cloverfield    Harrison Ford   
12                                    Sisu    Harrison Ford   
13                                   Trust    Harrison Ford   
14                       Spies in Disguise    Harrison Ford   
15             The Man with the Golden Gun    Harrison Ford   
16                          The Other Guys    Harrison Ford   
17                         Lethal Weapon 4    Harrison Ford   
18                    Terminator Salvation    Harrison Ford   
19                                     War    Harrison Ford   
20                                   Shaft    Harrison Ford   
21                      Terminator Genisys    Harrison Ford   
22                           The Dead Pool    Harrison Ford   
23         The Divergent Series: Insurgent    Harrison Ford   
24                                     War    Harrison Ford   
25                               Ong Bak 2    Harrison Ford   
26                                    Push    Harrison Ford   
27                             Van Helsing    Harrison Ford   
28                                   Shaft    Harrison Ford   
29                                     K-9    Harrison Ford   
30                         Another 48 Hrs.    Harrison Ford   
31                             Air America    Harrison Ford   
32                                 Ben-Hur    Harrison Ford   
33                        Charlie's Angels    Harrison Ford   
34                               MacGruber    Harrison Ford   
35          Gunjan Saxena: The Kargil Girl    Harrison Ford   
36       Spy Kids 2: Island of Lost Dreams    Harrison Ford   
37                        Charlie's Angels    Harrison Ford   
38                                 Elektra  Tatsuya Nakadai   
39                         Howard the Duck    Natsuki Hanae   
40               Kisi Ka Bhai Kisi Ki Jaan   Manoj Bajpayee   
41                              The Spirit          Prabhas   
42                             Zoolander 2   Toshirô Mifune   
43                                Snatched    Vicky Kaushal   

                  star2      votes     genre1  popularity certification  \
0           Karen Allen  1,003,504     Action         8.4             A   
1           Karen Allen    484,102     Action         8.1            UA   
2           Karen Allen    698,050     Action         7.9             A   
3           Karen Allen    353,843     Action         7.8           U/A   
4           Karen Allen    269,497     Action         7.6            UA   
5           Karen Allen     56,259     Action         7.5          None   
6           Karen Allen     27,866     Action         7.4          None   
7           Karen Allen     84,689     Action         7.3             U   
8           Karen Allen     29,659     Action         7.2             A   
9           Karen Allen    141,902     Action         7.1             A   
10          Karen Allen     29,659     Action         7.2             A   
11          Karen Allen    278,258     Action         7.0             A   
12          Karen Allen    478,773     Action         6.9             U   
13          Karen Allen     29,557     Action         6.9          None   
14          Karen Allen    113,857     Action         6.8             A